### ISSUE TITLE
Seems not working for 2.8 anymore

### ISSUE THREAD

**Issue by ameaninglessname** [khang06/mhynot2#7]
Note, I am running the tool and the game in [Sandboxie](https://github.com/sandboxie-plus/Sandboxie):

When I run this [console application](https://gist.github.com/khang06/56e3c221769648132023daab9fd2bc39) directly,
game boot up and shutdown within several seconds.

But if I "select something" in the `game's` console window to pause it, like :
![image](https://user-images.githubusercontent.com/20263103/182660775-21a9babe-212f-4bf2-94a9-b39f77b7e2b6.png)

The game boot into login screen and would let me do "click anything to enter the game",
But when the game is loading, it pops up a little window and say "Error happen, please go to the official website to download the client".

If you're running well, feel free to close this issue, then it should be something changed in recent Sandboxie's release.

**Comment by khang06** [comment]
it's possible that the game loaded mhyprot3 instead of 2, which isn't handled at all by this. i'm still not really sure about what criteria has to be met for one to be loaded instead of the other

**Comment by ameaninglessname** [comment]
> 

I am willing to test both ways.🤣

**Comment by ameaninglessname** [comment]
Would you update it recently?

**Comment by khang06** [comment]
probably not, mhyprot2 isn't even in the latest versions so it would be better to port it over to mhyprot3, but i don't think i'll do that for a while

**Comment by NullExir** [comment]
@ameaninglessname
Can you please specify which game version that was working with this bypass?, I've tried running v2.6 & 2.8 without a luck, game boots into main screen then suddenly shuts down, downloading v2.5 rn.
running win10 pro.

**Comment by ameaninglessname** [comment]
> @ameaninglessname
> Can you please specify which game version that was working with this bypass?, I've tried running v2.6 & 2.8 without a luck, game boots into main screen then suddenly shuts down, downloading v2.5 rn.
> running win10 pro.

It worked successfully in 2.7 version.

**Comment by NullExir** [comment]
@ameaninglessname
Thanks, i found out that we can run both 2.8, and 3.0
You'll need to halt the command line spawned by `libwinpthread-1.dll`, by selecting an are within the cmd.

Video proof:
[![proof](https://img.youtube.com/vi/fNYxC5XYBTA/maxresdefault.jpg)](https://youtu.be/fNYxC5XYBTA)
am using a server emulation software
@khang06 might be interested to simulate that behavior, either way appreciate the efforts :)

**Comment by ameaninglessname** [comment]
> `libwinpthread`

Thanks a lot!!! I will try it out

**Comment by ameaninglessname** [comment]
After tried several times, i didn't launch the game successfully.

Looks like the Chinese version and the International version has slightly different "anti mechanism" and naming.
![image](https://user-images.githubusercontent.com/20263103/188273214-ee472404-bc74-4a8e-9f7c-31ad4fcc1b20.png)

I also did two test:

- rename "mhyprot2.Sys" to "mhyprot2.Sys1",(make it view as deleted or missing by the YuanShen.exe)

and re-launch. the game crashed immediately.

- rename "mhyprot3.Sys" to "mhyprot3.Sys1",(make it view as deleted or missing by the YuanShen.exe)
and re-launch. the game window opened, by then crashed just like a "normal launch".

**Comment by NullExir** [comment]
@ameaninglessname You'll need to inject both `mhynot2` and `libwinpthread-1.dll` (from mingw lib folder), see the proof video above.
You can acquire that file from either MSYS2 or by installing mingw within a linux
here's a [link for both files compiled](https://drive.google.com/file/d/1Lph37prK2WZfVfC5a6xzhpAnzkXBMs0d/view?usp=sharing)

Are you using MSVC compiler ? try to compile in GCC (i compiled them inside a linux VM)
also see the unmerged pull request by **Jasuf** they made improvement to the `CMakeLists.txt`

**Comment by ameaninglessname** [comment]
Thanks for you files, but i still can't launch the game.
By selecting an area within the cmd the game is freezed, I have to press enter to continue running it.
(I remember that it could continue running at previous versions)

Yes, I compile both the dll and launcher(ConsoleApplication.exe) using MSVC, I don't have a linux VM.

![image](https://user-images.githubusercontent.com/20263103/188279314-794ddc81-5830-4678-a391-1889405ccea8.png)

**Comment by ameaninglessname** [comment]
Before tried your files, I also tried to inject both dll using my compiled stuff, (I copied the `libwinpthread-1.dll` from git installation folder, this dll is not a self compiled one right?) and failed.

**Comment by ameaninglessname** [comment]
Never mind, I can still play it on my cell phone, though the desktop graphic is more appealing.

**Comment by NullExir** [comment]
the launcher is looking specifically for `GenshinImpact.exe` [here](https://github.com/khang06/mhynot2/blob/be25d340b174f0b2d5791d160e485cd04316c4c5/launcher/launcher.cpp#L70) that might be the issue.

Try to recompile it with the correct process name for the your game version `YuanShen.exe`.
You can also try and download the [international version of the game](https://www.reddit.com/r/Genshin_Impact/comments/wuu5i9/direct_download_links_for_genshin_impact_30_from) see if it works.

**Comment by ameaninglessname** [comment]
> 

Yes, I have already changed the name there, that's the reason I have to use the "ConsoleApplication.exe" compiled myself, otherwise I can't launch the game at all.

I learned that the account of the Chinese version can't be logged into the international version, so even if it's currently work, it makes no sense for me.
